### PR TITLE
Add environment support for all deployable services

### PR DESCRIPTION
**Pull request by andytson-inviqa**
also provide the same level of support to service templates previously used only on application pod templates with valueFrom and envFrom

### CODE REVIEW

File: helm/app/templates/_base_helper.tpl [hunk 1]
```diff
@@ -87,6 +87,32 @@ stringData:
...
 
+{{- define "service.environment" }}
+{{- with (mergeOverwrite (dict) .service.environment .service.environment_dynamic)}}
+env:
+{{- range $key, $value := . }}
+- name: {{ $key | quote }}
+{{- if kindIs "string" $value }}
+  value: {{ tpl $value $.root | quote }}
+{{- else if kindIs "map" $value }}
+  {{ tpl ($value | toYaml) $.root | nindent 10 }}
```
Comment (kierenevans) [inline helm/app/templates/_base_helper.tpl]:
```suggestion
  {{ tpl ($value | toYaml) $.root | nindent 2 }}
```
To account for having nindent applied afterwards

File: helm/app/templates/_base_helper.tpl [hunk 2]
```diff
@@ -87,6 +87,32 @@ stringData:
...
+{{- end }}
+{{- end }}
+{{- if or .service.environment_secrets .service.envFrom }}
+envFrom:
+  {{- if .service.environment_secrets }}
+  - secretRef:
+      name: {{ print $.Release.Name "-" .serviceName }}
+  {{- end }}
+  {{- with .service.envFrom }}
+  {{- tpl (. | toYaml) $ | nindent 10 }}
```
Comment (kierenevans) [inline helm/app/templates/_base_helper.tpl]:
```suggestion
  {{- tpl (. | toYaml) $ | nindent 2 }}
```
To account for having nindent applied afterwards

File: helm/app/templates/_base_helper.tpl (hunk 1 above)
Comment (andytson-inviqa) [reply helm/app/templates/_base_helper.tpl]: thanks, yes sorry

though yaml doesn't care about indenting consistency between different object sub-values, but it'd look weird anyway

**Comment by andytson-inviqa** [comment]
thought I'd found a bug when environment rendering empty, but no, just adds a empty indented line

static-all-services tests that on redis for example

**Review by kierenevans — APPROVED** [review]

**Pull request merged**